be/src/exprs/agg/java_udaf_function.h

```diff
@@ -0,0 +1,194 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#pragma once
+
+#include <cstring>
+#include <memory>
+#include <vector>
+
+#include "column/binary_column.h"
+#include "exprs/agg/aggregate.h"
+#include "gen_cpp/Opcodes_types.h"
+#include "jni.h"
+#include "udf/java/java_udf.h"
+#include "udf/udf.h"
+#include "udf/udf_internal.h"
+
+namespace starrocks::vectorized {
+
+struct JavaUDAFState {
+    JavaUDAFState(jobject&& handle_) : handle(std::move(handle_)){};
+    ~JavaUDAFState() = default;
+    // UDAF State
+    jobject handle;
+};
+template <bool handle_null>
+jvalue cast_to_jvalue(MethodTypeDescriptor method_type_desc, const Column* col, int row_num);
+void release_jvalue(MethodTypeDescriptor method_type_desc, jvalue val);
+void append_jvalue(MethodTypeDescriptor method_type_desc, Column* col, jvalue val);
+Status get_java_udaf_function(int fid, const std::string& url, const std::string& checksum, const std::string& symbol,
+                              starrocks_udf::FunctionContext* context, AggregateFunction** func);
+// Not Support Nullable
+
+template <bool handle_null = true>
+class JavaUDAFAggregateFunction final : public AggregateFunction {
```

Comment:
Why template is `handle_null`, but comment is `Not Support Nullable`?

Reply:
Now FE Not support define no-nullable input function, we need support no-nullable input function in FE.
then we will create a UDAF Function like `sum`.we will return MakeNullable(new JavaUDAFAggregateFunction<false>()) if input was nullable and return new JavaUDAFAggregateFunction<false>() if input was no-nullable